src/ChemicalDiffusion/ChemicalDiffusion.jl

```diff
@@ -0,0 +1,152 @@
+    Buffer::Ptr{UInt8}  # Buffer condition (e.g., NNO) during the experiment
+    fluid::Ptr{UInt8}  # Fluid condition (e.g., anhydrous) during the experiment
+
+    function DiffusionData(;
+        Name="",  # name of the diffusion experiment and paper
+        Mineral="",  # name of the mineral
+        Formula="",  # chemical formula of the mineral
+        Species="",  # element or species being diffused
+        D0=0.0m^2/s,  # pre-exponential factor
+        D0_2σ=0.0m^2/s,  # uncertainty at 2σ of the pre-exponential factor
+        Ea=0.0J/mol,  # activation energy
+        Ea_2σ=0.0J/mol,  # uncertainty at 2σ of the activation energy
+        ΔV=0.0cm^3/mol,  # activation volume
+        ΔV_2σ=0.0cm^3/mol,  # uncertainty at 2σ of the activation volume
+        R=Unitful.R,  # gas constant
+        T_range=(0.0K, 0.0K),  # temperature range
+        P0=0.0Pa,  # pressure of calibration
+        Orientation="",  # Crystal orientation from the diffusion experiment
+        Crystallography="",  # Crystallographic system of the mineral
+        Buffer="",  # Buffer condition (e.g., NNO) during the experiment
+        Fluid=""  # Fluid condition (e.g., anhydrous) during the experiment
+    )
+
+        # Convert to GeoUnits
+        D0U = convert(GeoUnit, D0)
+        D0_2σU = convert(GeoUnit, D0_2σ)
+        EaU = convert(GeoUnit, Ea)
+        Ea_2σU = convert(GeoUnit, Ea_2σ)
+        ΔVU = convert(GeoUnit, ΔV)
+        ΔV_2σU = convert(GeoUnit, ΔV_2σ)
+        RU = convert(GeoUnit, R)
+        T_rangeU = convert.(GeoUnit, T_range)
+        P0U = convert(GeoUnit, P0)
+        # Extract struct types
+        T = typeof(D0U).types[1]
+        U1 = typeof(D0U).types[2]
+        U2 = typeof(EaU).types[2]
+        U3 = typeof(ΔVU).types[2]
+        U4 = typeof(RU).types[2]
+        U5 = typeof(T_rangeU[1]).types[2]
+        U6 = typeof(P0U).types[2]
+        name = pointer(ptr2string(Name))
+        mineral = pointer(ptr2string(Mineral))
+        formula = pointer(ptr2string(Formula))
+        species = pointer(ptr2string(Species))
+        orientation = pointer(ptr2string(Orientation))
+        crystallography = pointer(ptr2string(Crystallography))
+        buffer = pointer(ptr2string(Buffer))
+        fluid = pointer(ptr2string(Fluid))
+
+        # Create struct
+        return new{T,U1,U2,U3,U4,U5,U6}(
+            name, mineral, formula, species, D0U, D0_2σU, EaU, Ea_2σU, ΔVU, ΔV_2σU, RU, T_rangeU, P0U, orientation, crystallography, buffer, fluid
+        )
+    end
+end
+
+function param_info(data::DiffusionData) # info about the struct
+    return MaterialParamsInfo(;
+        Equation=L"D = D0 * \exp\left(-\frac{Ea + PΔV} {RT}\right)",
+    )
+end
+
+# Calculation routines for the diffusion coefficient
+@inline function compute_D(data::DiffusionData; T=1K, P=0Pa, kwargs...)
+    @unpack_units D0, Ea, ΔV, P0, R = data
+
+    D = @pow D0 * exp(- (Ea + (P - P0) * ΔV) / (R * T))
```

Comment:
`@pow` is used to provide faster powers for floating point powers. Here (and also in some functions below) there are not even powers you can actually remove it.